duo-client/src/main/java/com/duosecurity/client/Http.java

```diff
@@ -313,6 +321,27 @@ private String canonQueryString()
     return Util.join(args.toArray(), "&");
   }
 
+  private String canonXDuoHeaders(){
+    ArrayList<String> canonList = new ArrayList<String>();
+    for (String name : additionalHeaders.keySet()){
+      String value = additionalHeaders.get(name);
+      canonList.add(name + value);
+    }
+    return Util.join(canonList.toArray(), String.valueOf(Character.MIN_VALUE));
+  }
+
+  private void validateAdditionalHeader(String name, String value) throws IllegalArgumentException{
+    if (name == null || name.length() == 0){
+      throw new IllegalArgumentException("Not allowed 'Null' or empty header name");
+    } else if (value == null || value.length() == 0){
+      throw new IllegalArgumentException("Not allowed 'Null' or empty header value");
+    } else if (!name.toLowerCase().startsWith("x-duo-")){
+      throw new IllegalArgumentException("Additional headers must start with \'X-Duo-\'");
+    } else if (additionalHeaders.containsKey(name)){
```

Comment:
what is the rationale for erroring if a header name is provided twice?  (side note, it's totally permitted in http for a header to have multiple values)

Reply:
We raise errors in python client, that's the main reason I raise exception here